### ISSUE TITLE
Crate no longer compiles on arm

### ISSUE THREAD

**Issue by petevine** [Stebalien/tempfile#18]
Breakage caused by [this](https://github.com/rust-lang/rust/issues/29867).

````
 Compiling tempfile v1.1.1
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/tempfile-1.1.1/src/imp/unix.rs:23:20: 23:46 error: mismatched types:
 expected `*const i8`,
    found `*const u8`
(expected i8,
    found u8) [E0308]
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/tempfile-1.1.1/src/imp/unix.rs:23         libc::open(try!(cstr(&path)).as_ptr(), O_CLOEXEC | O_EXCL | O_RDWR | O_CREAT, 0o600)
                                                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/tempfile-1.1.1/src/imp/unix.rs:23:20: 23:46 help: run `rustc --explain E0308` to see a detailed explanation
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/tempfile-1.1.1/src/imp/unix.rs:35:20: 35:44 error: mismatched types:
 expected `*const i8`,
    found `*const u8`
(expected i8,
    found u8) [E0308]
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/tempfile-1.1.1/src/imp/unix.rs:35         libc::open(try!(cstr(dir)).as_ptr(), O_CLOEXEC | O_EXCL | O_TMPFILE | O_RDWR, 0o600)
                                                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/tempfile-1.1.1/src/imp/unix.rs:35:20: 35:44 help: run `rustc --explain E0308` to see a detailed explanation
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/tempfile-1.1.1/src/imp/unix.rs:95:24: 95:54 error: mismatched types:
 expected `*const i8`,
    found `*const u8`
(expected i8,
    found u8) [E0308]
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/tempfile-1.1.1/src/imp/unix.rs:95             libc::open(try!(cstr(&tmp_path)).as_ptr(), O_CLOEXEC | O_EXCL | O_RDWR | O_CREAT, 0o600)
                                                                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/odroid/.cargo/registry/src/github.com-48ad6e4054423464/tempfile-1.1.1/src/imp/unix.rs:95:24: 95:54 help: run `rustc --explain E0308` to see a detailed explanation
error: aborting due to 3 previous errors
````

**Comment by Stebalien** [comment]
Well, that's mildly infuriating. Does the fix-arm branch work?

**Comment by petevine** [comment]
The fix works, thank you. And yes it's mighty infuriating!

**Comment by Stebalien** [comment]
Fixed in e7e800a9a3e7ccc966d81158c78b33ac27babbf7

**Comment by Stebalien** [comment]
Also released 1.1.3
